src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -209,6 +215,23 @@ private async Task SearchCurrentDocumentAsync(CancellationToken cancellationToke
             }
         }
 
+        private async Task ProcessResultsAsync(
+            HashSet<INavigateToSearchResult> seenItems,
+            NavigateToResultStream results,
+            CancellationToken cancellationToken)
+        {
+            await foreach (var (project, result) in results)
+            {
+                // If we're seeing a dupe in another project, then filter it out here.  The results from
+                // the individual projects will already contain the information about all the projects
+                // leading to a better condensed view that doesn't look like it contains duplicate info.
+                if (!seenItems.Add(result))
```

Comment:
note: we no longer need a lock as this is always only accessed from one thread at a time.  a nice fallout of pullin ginstead of pushing.